Fix issue 27162: multi-keyword display is everywhere now (#29103)
mdn/content: files/en-us/web/css/display/multi-keyword_syntax_of_display/index.md

{{CSSRef}}

[CSS Display Module Level 3](https://drafts.csswg.org/css-display/) describes the multi-keyword syntax for the [`display`](/en-US/docs/Web/CSS/display) property.

> This syntax is supported in the current versions of all major browsers. This guide explains the new syntax, including the reasoning behind this change. It also details the in-built backwards compatibility for the `display` property.

Reviewer: I think we can get rid of this entire paragraph, since it will be outdated soon if not yesterday. Or, maybe change to something like:

```suggestion
Supported in all major browsers, this guide explains the "two-value syntax" or "multi-value syntax syntax, and the historical evolution from single-value syntax.
```

Which could be appended to the first paragraph, and the next note is no longer needed.

> [CSS Display Module Level 3](https://drafts.csswg.org/css-display/) describes the multi-keyword syntax for the [`display`](/en-US/docs/Web/CSS/display) property.
> 
> This syntax is supported in the current versions of all major browsers. This guide explains the new syntax, including the reasoning behind this change. It also details the in-built backwards compatibility for the `display` property.

Author: I'm happy to remove the "supported in all major browsers", bit but I like the bit introducing what this guide does. Could we say just:

```suggestion
[CSS Display Module Level 3](https://drafts.csswg.org/css-display/) introduced multi-keyword syntax for the [`display`](/en-US/docs/Web/CSS/display) property. This guide explains the multi-keyword syntax.
```

?

> As you can see from the above explanation, the `display` property has gained considerable new powers. In addition to indicating whether something is block-level or inline-level in relationship to other boxes on the page, it now also indicates the formatting context inside the box it is applied to. In order to better describe this behavior, the `display` property has been refactored to allow for two values — an outer and inner value — to be set on it, as well as the single values we are used to.

Reviewer: ```suggestion
As you can see from the above explanation, the `display` property has considerable powers. In addition to indicating whether something is block-level or inline-level in relationship to other boxes on the page, it also indicates the formatting context inside the box it is applied to. To better describe this behavior, the `display` property allows for two values — an outer and inner value — to be set on it. The original single-value syntax is also valid.
```

> This means that instead of setting `display: flex` to create a block-level box with flex children, we will use `display: block flex`. Instead of `display: inline-flex` to create an inline-level box with flex children, we will use `display: inline flex`. The example below values.

Reviewer: ```suggestion
This means that instead of setting `display: flex` to create a block-level box with flex children, we use `display: block flex`. Instead of `display: inline-flex` to create an inline-level box with flex children, we use `display: inline flex`. The example below values.
```

Author: (stealth edit to fix a typo I had introduced)

> In terms of how these new values help to clarify CSS layout, we can take a look at a couple of values in the table that might seem less familiar. The multi-keyword `display: block flow-root` maps to a fairly recent single value; `display: flow-root`. This value's only purpose is to create a new [Block Formatting Context](/en-US/docs/Web/Guide/CSS/Block_formatting_context) (BFC). A BFC ensures that everything inside your box stays inside it, and things from outside the box cannot intrude into it. The most obvious use-case for creating a new BFC is to contain floats, and avoid the need for clearfix hacks.

Reviewer: ```suggestion
In terms of how this multi-value syntax helps clarify CSS layout, we can take a look at a couple of values in the table that might seem less familiar. The multi-keyword `display: block flow-root` maps to a relatively recent single value; `display: flow-root`. This value's only purpose is to create a new [Block Formatting Context](/en-US/docs/Web/Guide/CSS/Block_formatting_context) (BFC). A BFC ensures that everything inside your box stays inside it, and things from outside the box cannot intrude into it. When `flow-root` was added as a `display` value, a common use case for creating a new BFC was to contain floats and avoid the need for clearfix hacks.
```

{{EmbedGHLiveSample("css-examples/display/multi-keyword/inline-block.html", '100%', 440)}}

> An element with `display: inline-block` however, will also contain floats. It contains everything inside the inline-level box. Therefore `display: inline-block` does exactly what `display: flow-root` does, but with an inline-level, rather than a block-level box. The new syntax accurately describes what is happening with this value. In the example above, you can change `display: inline-block` to `display: inline flow-root` and get the same result.

Reviewer: ```suggestion
An element with `display: inline-block` however, will also contain floats. It contains everything inside the inline-level box. Therefore `display: inline-block` does exactly what `display: flow-root` does, but with an inline-level, rather than a block-level box. The two-value syntax accurately describes what is happening with this value. In the example above, you can change `display: inline-block` to `display: inline flow-root` and get the same result.
```
